jenkins/bootstrap.py

```diff
@@ -303,6 +306,11 @@ def finish(gsutil, paths, success, artifacts, build, version, repo):
         except subprocess.CalledProcessError:
             logging.warning('Failed to write build path to %s', paths.pr_path)
 
+    meta = metadata(repo, artifacts)
+    if not version:
+        version = meta.get('job-version')
```

Comment:
I have a PR in kubernetes to handle this. https://github.com/kubernetes/kubernetes/pull/36654